test/parse/node.test.js

```diff
@@ -1,147 +1,182 @@
+			const dom = new DOMParser().parseFromString('<xml />')
+			const fragment = dom.createDocumentFragment()
+			expect(fragment.nodeType).toBe(Node.DOCUMENT_FRAGMENT_NODE)
+			const first = fragment.appendChild(dom.createElement('first'))
+			const last = fragment.appendChild(dom.createElement('last'))
+			expectNeighbours(first, last)
+			expect(fragment.firstChild).toStrictEqual(first)
+			expect(fragment.lastChild).toStrictEqual(last)
+		})
 	})
 
-	it('insert node', () => {
-		var dom = new DOMParser().parseFromString('<xml><child/></xml>')
-		var node = dom.createElement('sibling')
-		var child = dom.documentElement.firstChild
-		child.parentNode.insertBefore(node, child)
-		assert(node, child.previousSibling)
-		assert(node.nextSibling, child)
-		assert(node.parentNode, child.parentNode)
-	})
+	describe('insertBefore', () => {
+		it('places created element before existing element', () => {
+			const dom = new DOMParser().parseFromString('<xml><child/></xml>')
+			const inserted = dom.createElement('sibling')
+			const child = dom.documentElement.firstChild
+
+			child.parentNode.insertBefore(inserted, child)
+
+			expectNeighbours(inserted, child)
+		})
+		it('inserts all elements in DocumentFragment', () => {
+			const dom = new DOMParser().parseFromString('<xml><child/></xml>')
+			const child = dom.documentElement.firstChild
 
-	it('insert fragment', () => {
-		var dom = new DOMParser().parseFromString('<xml><child/></xml>')
-		var fragment = dom.createDocumentFragment()
-		assert(fragment.nodeType, 11)
-		var first = fragment.appendChild(dom.createElement('first'))
-		var last = fragment.appendChild(dom.createElement('last'))
-		assert(fragment.firstChild, first)
-		assert(fragment.lastChild, last)
-		assert(last.previousSibling, first)
-		assert(first.nextSibling, last)
-		var child = dom.documentElement.firstChild
-		child.parentNode.insertBefore(fragment, child)
-		assert(last.previousSibling, first)
-		assert(first.nextSibling, last)
-		assert(child.parentNode.firstChild, first)
-		assert(last, child.previousSibling)
-		assert(last.nextSibling, child)
-		assert(first.parentNode, child.parentNode)
-		assert(last.parentNode, child.parentNode)
+			const fragment = dom.createDocumentFragment()
+			const first = fragment.appendChild(dom.createElement('first'))
+			const second = fragment.appendChild(dom.createElement('second'))
+
+			child.parentNode.insertBefore(fragment, child)
+
+			expectNeighbours(first, second, child)
+			expect(child.parentNode.firstChild).toStrictEqual(first)
+		})
 	})
 
-	it('instruction', () => {
-		var source =
+	it('preserves instruction', () => {
+		const source =
 			'<?xml version="1.0"?><root><child>&amp;<!-- &amp; --></child></root>'
-		var doc = new DOMParser().parseFromString(source, 'text/xml')
-		var source2 = new XMLSerializer().serializeToString(doc)
-		assert(source2, source)
+		expect(new DOMParser().parseFromString(source, 'text/xml').toString()).toBe(
+			source
+		)
```

Comment:
last nit for now (hopefuly): I find the formatting of this statement to be a bit ugly. I would do something like this:

```suggestion
		const output = new DOMParser().parseFromString(source, 'text/xml').toString()
		expect(output).toBe(source)
```

Reply:
:+1: I will also check if there are other places where "arrange, act, assert" are not clearly split by (empty) lines.